### ISSUE TITLE
Fault mitigation would not work in interrupt handler. 

### ISSUE THREAD

**Issue by sichunqin** [OP-TEE/optee_os#6309]
I try to add optee fault mitigation measure to one interrupt handler function, but fail due to the following error: 

[TEE] E/TC:1    assertion 'ct >= 0 && ct < CFG_NUM_THREADS' failed at core/arch/arm/kernel/thread.c:799 <thread_get_id>_

The call chain is __ftmn_get_tsd_func_arg_pp=>thread_get_tsd=>thread_get_id. It seems that interrupt handler is not associated with any thread, so the ct (current_thread_id) is -1.

Interrupt handler function would be attacked, and should be protected. Any idea to make it work? 

```
static inline struct ftmn_func_arg **__ftmn_get_tsd_func_arg_pp(void)
{
#if defined(CFG_FAULT_MITIGATION) && defined(__KERNEL__)
	return &thread_get_tsd()->ftmn_arg;
#elif defined(CFG_FAULT_MITIGATION)
	return &__ftmn_global_func_arg;
#else
	return NULL;
#endif
} 

int thread_get_id(void)
{
	int ct = thread_get_id_may_fail();

	assert(ct >= 0 && ct < CFG_NUM_THREADS);
	return ct;
}

```

**Comment by sichunqin** [comment]
One simple fix would be use  __ftmn_global_func_arg which is already defined for no kernel scenario if thread_id < 0. 
I am not sure if this is proper fix. 

-#ifndef __KERNEL__
+//#ifndef __KERNEL__
 extern struct ftmn_func_arg *__ftmn_global_func_arg;
-#endif
+//#endif

static inline struct ftmn_func_arg **__ftmn_get_tsd_func_arg_pp(void)
 {
 #if defined(CFG_FAULT_MITIGATION) && defined(__KERNEL__)
-       return &thread_get_tsd()->ftmn_arg;
+    if (thread_get_id_may_fail() >= 0)
+               return &thread_get_tsd()->ftmn_arg;
+       else
+               return &__ftmn_global_func_arg;
 #elif defined(CFG_FAULT_MITIGATION)
        return &__ftmn_global_func_arg;
 #else

**Comment by jenswi-linaro** [comment]
You should probably place the extra `struct ftmn_func_arg` pointer in `struct thread_core_local` instead.
Feel free to create a pull request and we can continue the discussion there.

**Comment by sichunqin** [comment]
Thanks for the quick reply. PR is created at https://github.com/OP-TEE/optee_os/pull/6312.

**Comment by sichunqin** [comment]
Fixed by PR #6312.
